fn/handler.go

```diff
@@ -80,8 +78,12 @@ func fromMethods(rcvr interface{}, t reflect.Type) rpc.Handler {
 	return mux
 }
 
+var callRef = reflect.TypeOf((*rpc.Call)(nil))
+
 func fromFunc(fn reflect.Value) rpc.Handler {
 	fntyp := fn.Type()
+	// if the last argument in fn is an rpc.Call, add our call to fnParams
+	expectsCallParam := fntyp.NumIn() > 0 && fntyp.In(fntyp.NumIn()-1) == callRef
 
 	return rpc.HandlerFunc(func(r rpc.Responder, c *rpc.Call) {
 		defer func() {
```

Comment:
I think we also want to do this in Call, converting panics to errors. But also keep this one here as well.